contrib/gocql/gocql/gocql.go

```diff
@@ -35,6 +35,12 @@ type params struct {
 }
 
 // WrapQuery wraps a gocql.Query into a traced Query under the given service name.
+// Note that the returned Query structure embeds the original gocql.Query structure.
+// This means that any method returning the query for chaining that is not part
+// of this package's Query structure should be called before WrapQuery, otherwise
+// the tracing context could be lost.
+// For example, it is ok to use and chain the return value of `WithContext` and
+// `PageState` but not that of `Consistency`, `Trace`, `Observer`, etc.
```

Comment:
I would also emphasise that `WithContext` must be called for `Query` and not `gocql.Query` to ensure that inheritance is properly chained.

Reply:
Agreed. I rephrased a little hinting towards that.